test/conformance/runtime/util.go

```diff
@@ -31,17 +31,33 @@ import (
 
 // fetchRuntimeInfoUnprivileged creates a Service that uses the 'runtime-unprivileged' test image, and extracts the returned output into the
 // RuntimeInfo object.
-func fetchRuntimeInfoUnprivileged(t *testing.T, clients *test.Clients, opts ...ServiceOption) (*test.ResourceNames, *types.RuntimeInfo, error) {
-	return runtimeInfo(t, clients, &test.ResourceNames{Image: test.RuntimeUnprivileged}, opts...)
+func fetchRuntimeInfoUnprivileged(
+	t *testing.T,
+	clients *test.Clients,
+	reqOpts []pkgTest.RequestOption,
```

Comment:
I'd go with opts ...interface{}

and the in the actual implementation you can loop and separate.
```
for _, opt := range opts {
switch t := opt.(type) {
...
}
}
```
}
This way you can avoid all these pesky nils.

Reply:
this is what i want.